lib/filters/http.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+var preq = require('preq');
+var utils = require('../utils');
+
+module.exports = function(hyper, req, next, options) {
+    options = options || {};
+    var host = req.uri.constructor === String ? req.uri : req.uri.protoHost;
+    var match;
+    if (options.allow) {
```

Comment:
It might be worth caching the result of this in something like `options._cache.matcher`, and then simply run it if set up.